src/js/angular/core/directives/yasgui-component/yasgui-component.directive.js

```diff
@@ -0,0 +1,624 @@
+                    backdrop: 'static',
+                    scope: progressScope
+                });
+                return connectorProgressModal;
+            };
+
+            const toCreateCommandResponse = (response, tabId) => {
+                const connectorProgressModal = createConnectorProgressDialog($translate.instant('externalsync.creating'), response.data.iri, response.data.name);
+                tabIdToConnectorProgressModalMapping.set(tabId, connectorProgressModal);
+                return new BeforeUpdateQueryResult(BeforeUpdateQueryResultStatus.SUCCESS, 'created.connector', getCommandParameters(response));
+            };
+
+            const toRepairCommandResponse = (response, tabId) => {
+                const connectorProgressModal = createConnectorProgressDialog($translate.instant('externalsync.repairing'), response.data.iri, response.data.name);
+                tabIdToConnectorProgressModalMapping.set(tabId, connectorProgressModal);
+                return new BeforeUpdateQueryResult(BeforeUpdateQueryResultStatus.SUCCESS, 'query.editor.repaired.connector', getCommandParameters(response));
+            };
+
+            const toDropCommandResponse = (response) => {
+                return new BeforeUpdateQueryResult(BeforeUpdateQueryResultStatus.SUCCESS, 'externalsync.delete.success.msg', getCommandParameters(response));
+            };
+
+            const getBeforeUpdateQueryHandler = () => (query, tabId) => {
+                return new Promise(function (resolve) {
+                    ConnectorsRestService.checkConnector(query)
+                        .then((response) => {
+                            if (!response.data.command) {
+                                resolve(toNoCommandResponse());
+                                return;
+                            }
+                            if (!response.data.hasSupport) {
+                                resolve(toHasNotSupport(response));
+                                return;
+                            }
+                            switch (response.data.command) {
+                                case 'create':
+                                    resolve(toCreateCommandResponse(response, tabId));
+                                    break;
+                                case 'repair':
+                                    resolve(toRepairCommandResponse(response, tabId));
+                                    break;
+                                case 'drop':
+                                    resolve(toDropCommandResponse(response));
+                                    break;
+                            }
+                        }).catch(() => {
+                        // for some reason we couldn't check if this is a connector update, so just execute it
+                        resolve();
+                    });
+                });
+            };
+
+            const exploreVisualGraphYasrToolbarElementBuilder = {
```

Comment:
Bot sure if this should be here or at least we need to generalize it a bit and also rename it. Although, this directive is supposed to be kind of generic component.

Reply:
Moved as configuration to the sparql view